homeassistant/components/channels/media_player.py

```diff
@@ -113,106 +100,41 @@ def update_favorite_channels(self):
-            self.channel_name = channel_hash.get("channel_name")
-            self.channel_image_url = channel_hash.get("channel_image_url")
-        else:
-            self.channel_number = None
-            self.channel_name = None
-            self.channel_image_url = None
-
+            channel_image_url = channel_hash.get("channel_image_url")
+        now_playing_title = now_playing_image_url = None
         if np_hash:
-            self.now_playing_title = np_hash.get("title")
-            self.now_playing_episode_title = np_hash.get("episode_title")
-            self.now_playing_season_number = np_hash.get("season_number")
-            self.now_playing_episode_number = np_hash.get("episode_number")
-            self.now_playing_summary = np_hash.get("summary")
-            self.now_playing_image_url = np_hash.get("image_url")
-        else:
-            self.now_playing_title = None
-            self.now_playing_episode_title = None
-            self.now_playing_season_number = None
-            self.now_playing_episode_number = None
-            self.now_playing_summary = None
-            self.now_playing_image_url = None
-
-    @property
-    def name(self):
-        """Return the name of the player."""
-        return self._name
-
-    @property
-    def state(self):
-        """Return the state of the player."""
+            now_playing_title = np_hash.get("title")
+            now_playing_image_url = np_hash.get("image_url")
+        self._attr_source_list = [channel["name"] for channel in self.favorite_channels]
+        self._attr_media_title = now_playing_title if self._attr_state else None
+        self._attr_media_image_url = (
+            now_playing_image_url
+            if now_playing_image_url
+            else channel_image_url
+            if channel_image_url
+            else "https://getchannels.com/assets/img/icon-1024.png"
+        )
+        self._attr_state = None
         if self.status == "stopped":
-            return STATE_IDLE
-
-        if self.status == "paused":
-            return STATE_PAUSED
-
-        if self.status == "playing":
-            return STATE_PLAYING
-
-        return None
+            self._attr_state = STATE_IDLE
+        elif self.status == "paused":
+            self._attr_state = STATE_PAUSED
+        elif self.status == "playing":
+            self._attr_state = STATE_PLAYING
 
     def update(self):
         """Retrieve latest state."""
         self.update_favorite_channels()
         self.update_state(self.client.status())
 
-    @property
-    def source_list(self):
-        """List of favorite channels."""
-        sources = [channel["name"] for channel in self.favorite_channels]
-        return sources
-
-    @property
-    def is_volume_muted(self):
-        """Boolean if volume is currently muted."""
-        return self.muted
-
-    @property
-    def media_content_id(self):
-        """Content ID of current playing channel."""
-        return self.channel_number
```

Comment:
I can't see `_attr_media_content_id` being set as a replacement. Is it not necessary?